File: server/object/dto/PostServerResponse.go
package dto

import (
	"promise/common/object/constValue"
	"promise/server/object/model"
)

// PostServerResponse is DTO.
type PostServerResponse struct {
	ResourceResponse
	ID           string `json:"ID"`
	URI          string `json:"URI"`
	PhysicalUUID string `json:"PhysicalUUID"`
	Hostname     string `json:"Hostname"`
	Type         string `json:"Type"`
}

// Load will load data from model.
func (dto *PostServerResponse) Load(m *model.Server) {
	dto.ID = m.ID
	dto.URI = constValue.ToServerURI(m.ID)
	dto.PhysicalUUID = m.PhysicalUUID
	dto.Hostname = m.Hostname
	dto.Type = m.Type
}

File: integration-test/src/test/java/com/promise/integrationtest/server/dto/PostServerResponse.java
package com.promise.integrationtest.server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.promise.integrationtest.base.ResourceResponse;

/**
 * DTO of post server response.
 *
 */
public class PostServerResponse extends ResourceResponse
{
    @JsonProperty(value = "Hostname", required = true)
    private String hostname;
    @JsonProperty(value = "Type", required = true)
    private String type;
    @JsonProperty(value = "PhysicalUUID", required = true)
    private String physicalUUID;

    public String getHostname()
    {
        return hostname;
    }

    public void setHostname(String hostname)
    {
        this.hostname = hostname;
    }

    public String getType()
    {
        return type;
    }

    public void setType(String type)
    {
        this.type = type;
    }

    public String getPhysicalUUID()
    {
        return physicalUUID;
    }

    public void setPhysicalUUID(String physicalUUID)
    {
        this.physicalUUID = physicalUUID;
    }
}
